nni/compression/pytorch/speedup/compressor.py

```diff
@@ -43,23 +45,18 @@ class ModelSpeedup:
         the index of batch dimension in the dummy_input
     confidence: the confidence coefficient of the sparsity inference. This value is
         actually used as the batchsize of the dummy_input.
-    customized_replace_func: None/Dict
-        If `customized_replace_func` is not None, then we will use the given function to replace the
-        corresponding modules. The `key` of the dict is the opertor types and the `value`
-        is the replace function of corresponding opertor. The replace function should take
-        two input parameters, one is the original module, the second input parameter is tuple
-        of the input mask, output mask and weight mask. This replace function should prune the module
-        accordingly. Here is an example of the replace function(more examples can refer to compress_modules.py)::
-
-            def example_replace(ori_module, masks):
-                in_mask, out_mask, weight_mask = masks
-                # prune the ori_module to a new smaller module according to the mask
-                return new_small_module
-
+    customized_replacers
+        ``customized_replacers`` is a list of ``Replacer``.
+        Call a ``Module`` that does not contain a ``Module`` as a leaf-module,
+        a ``Module`` that contains a ``Module`` as a hyper-module, then replacer is used to replace the hyper-module.
+        The difference between the replacer and replace function is that replacer can perform more efficient replacements
+        to hyper-module, and replace function is used to replace leaf-module.
+        In ``ModelSpeedup.compress``, replacers are first to be called to replace the hyper-modules before
+        replacing all leaf-modules by replace functions.
     """
 
     def __init__(self, model, dummy_input, masks_file, map_location=None,
-                 batch_dim=0, confidence=8, customized_replace_func=None):
+                 batch_dim=0, confidence=8, customized_replacers=None, customized_replace_func=None):
```

Comment:
might be better to add "customized_repalce_func" in docstring and add "deprecated" warning in it.